From 0f8fdfb55950e383d9780b875bc8d7213db8a42a Mon Sep 17 00:00:00 2001
From: Brad King <brad.king@kitware.com>
Date: Wed, 24 Sep 2025 12:02:37 -0400
Subject: [PATCH] bootstrap: Fix compilation with gcc 16 devirtualization

Avoid including a header for a subclass whose implementation is not
compiled during bootstrap.

Fixes: #27256

Upstream-Status: Backport [v3.31.10 ea04e19daf7010781d0df980b9683a642093e381 https://gitlab.kitware.com/cmake/cmake/-/merge_requests/11243]
Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
[Batocera: required on Arch host with GCC 16 for host-cmake 3.30.5 bootstrap;
 adapted whitespace to match cmake 3.30.5 Source/cmake.cxx]
---
 Source/cmake.cxx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Source/cmake.cxx b/Source/cmake.cxx
index 1111111..2222222 100644
--- a/Source/cmake.cxx
+++ b/Source/cmake.cxx
@@ -126,12 +126,12 @@
 #  include "cmExtraEclipseCDT4Generator.h"
 #  include "cmExtraKateGenerator.h"
 #  include "cmExtraSublimeTextGenerator.h"
-#endif
 
 // NOTE: the __linux__ macro is predefined on Android host too, but
 // main CMakeLists.txt filters out this generator by host name.
-#if (defined(__linux__) && !defined(__ANDROID__)) || defined(_WIN32)
-#  include "cmGlobalGhsMultiGenerator.h"
+#  if (defined(__linux__) && !defined(__ANDROID__)) || defined(_WIN32)
+#    include "cmGlobalGhsMultiGenerator.h"
+#  endif
 #endif
 
 #if defined(__APPLE__)
